x-pack/plugins/infra/public/pages/link_to/redirect_to_node_detail.tsx

```diff
@@ -38,33 +37,14 @@ export const RedirectToNodeDetail = ({
     }
   }
 
-  return <Redirect to={`/detail/${nodeType}/${nodeId}?${queryParams.toString()}`} />;
-};
-
-export const getNodeDetailUrl = ({
-  nodeType,
-  nodeId,
-  to,
-  from,
-  assetName,
-}: {
-  nodeType: InventoryItemType;
-  nodeId: string;
-  to?: number;
-  from?: number;
-  assetName?: string;
-}): LinkDescriptor => {
-  return {
-    app: 'metrics',
-    pathname: `link-to/${nodeType}-detail/${nodeId}`,
-    search: {
-      ...(assetName ? { assetName } : undefined),
-      ...(to && from
-        ? {
-            to: `${to}`,
-            from: `${from}`,
-          }
-        : undefined),
-    },
-  };
+  const state = getStateFromLocation(location);
+  return (
+    <Redirect
+      to={{
+        pathname: `/detail/${nodeType}/${nodeId}`,
+        search: queryParams.toString(),
+        state: state ? JSON.parse(state) : undefined,
```

Comment:
Might be worth wrapping `JSON.parse` with try/catch considering the unstructured nature of the `state` property.

Reply:
True! Good catch.